### PR TITLE
GUACAMOLE-699: Update german translations

### PR DESCRIPTION
**Pull request by leetxyz**
Updated the main german translation and added translations for the extensions.
I also switched the translation from a non-formal ("Du") to the formal ("Sie") expression.
Please note that i can't test the extension-translations, but checked them against a JSON-Validator.

Feel free to contact me through Jira for questions / clarifications (https://issues.apache.org/jira/browse/GUACAMOLE-699)

### CODE REVIEW

**Comment by mike-jumper** [comment]
Please also squash your commits if there is no logical change worth citing in the message. Repeating the same "GUACAMOLE-699: Add initial german translation" message seven times doesn't add any value to the history.

**Comment by necouchman** [comment]
@leetxyz: Wondering if you can fix up these issues so that we can merge these changes:
- Remove translation strings that are not actually translated
- Squash commits.

**Comment by leetxyz** [comment]
Dear all
Sorry for the delay, was on holiday.

- I squashed the commits
- I added the translations the guac-manifests

For the unneeded translations and @APP-Notations, could you please give me an example?

thanks and best regards

**Comment by necouchman** [comment]
@mike-jumper: Not sure on some of these what should be left and what should be removed...
- Lines for empty strings (fields with no headers, empty field options, etc.)?
- Lines that refer to other lines (`@:APP.PARAMETER`, etc.)?

**Comment by mike-jumper** [comment]
> Lines for empty strings (fields with no headers, empty field options, etc.)?

These should be kept if the empty string is the correct German translation for what is represented by an empty string in English (probably the case).

> Lines that refer to other lines (@:APP.PARAMETER, etc.)?

Same here - these should be kept if the translation as written is correct for German (assuming the line being referenced with `@:` exists in the new translation - angular-translate can't reference strings across translations).

The general rule is to omit strings which are untranslated because the translator couldn't or didn't translate them. If the string *has* been translated and just happens to be the same in both languages, then it should be duplicated. The current exceptions to this are `APP.NAME` and `APP.VERSION` which we only want defined in one location for simplicity and support for downstream branding.

File: guacamole/src/main/webapp/translations/de.json [hunk 1]
```diff
@@ -400,27 +534,56 @@
         "FIELD_OPTION_FONT_SIZE_96"    : "96",
         "FIELD_OPTION_FONT_SIZE_EMPTY" : "",
 
+        "FIELD_OPTION_TERMINAL_TYPE_ANSI"           : "ansi",
+        "FIELD_OPTION_TERMINAL_TYPE_EMPTY"          : "",
+        "FIELD_OPTION_TERMINAL_TYPE_LINUX"          : "linux",
+        "FIELD_OPTION_TERMINAL_TYPE_VT100"          : "vt100",
+        "FIELD_OPTION_TERMINAL_TYPE_VT220"          : "vt220",
+        "FIELD_OPTION_TERMINAL_TYPE_XTERM"          : "xterm",
+        "FIELD_OPTION_TERMINAL_TYPE_XTERM_256COLOR" : "xterm-256color",
```
Comment (necouchman) [inline guacamole/src/main/webapp/translations/de.json]: Are these accurate translations?

File: guacamole/src/main/webapp/translations/de.json [hunk 2]
```diff
@@ -400,27 +534,56 @@
...
+
         "NAME" : "SSH",
 
         "SECTION_HEADER_AUTHENTICATION" : "Authentifizierung",
+        "SECTION_HEADER_BEHAVIOR"       : "Terminal behavior",
```
Comment (necouchman) [inline guacamole/src/main/webapp/translations/de.json]: Translation, or copy?

File: extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/resources/translations/de.json [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        "NAME" : "Geteilte Verbindungen (SQL Server)"
+    },
+
+    "HOME" : {
+        "INFO_SHARED_BY" : "Geteilt mit {USERNAME}"
+    },
+
+    "PASSWORD_POLICY" : {
+
+        "ERROR_CONTAINS_USERNAME"      : "Passwörten dürfen Ihren Benutzername nicht enthalten.",
```
Comment (mike-jumper) [inline extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/resources/translations/de.json]: From @berndhose on #399:

> I have reviewed the proposed changes of #375 and they seem to be ok, apart from one occurrence of the word "Passwörten", which should read "Passwörter".

Comment (berndhose) [reply extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/resources/translations/de.json]: reviewed some more, see following messages

File: extensions/guacamole-auth-duo/src/main/resources/translations/de.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
+
+    "DATA_SOURCE_DUO" : {
+        "NAME" : "Duo TFA Backend"
```
[the remark this answers is not in the document]
Comment (berndhose) [reply extensions/guacamole-auth-duo/src/main/resources/translations/de.json]: translation is ok

File: extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/resources/translations/de.json [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        "FIELD_HEADER_GUACD_ENCRYPTION" : "Verschlüsselung:",
+        "FIELD_HEADER_GUACD_PORT"       : "Port:",
+
+        "FIELD_OPTION_GUACD_ENCRYPTION_EMPTY" : "",
+        "FIELD_OPTION_GUACD_ENCRYPTION_NONE"  : "Keine (unverschlüsselt)",
+        "FIELD_OPTION_GUACD_ENCRYPTION_SSL"   : "SSL / TLS",
+
+        "SECTION_HEADER_CONCURRENCY"    : "Verbindungslimite",
+        "SECTION_HEADER_LOAD_BALANCING" : "Lastverteilung",
+        "SECTION_HEADER_GUACD"          : "Guacamole Proxy Parameter (guacd)"
```
[the remark this answers is not in the document]
Comment (berndhose) [reply extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/resources/translations/de.json]: translation is ok

File: extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/resources/translations/de.json [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+        "FIELD_HEADER_ENABLE_SESSION_AFFINITY"  : "Sessionzugehörigkeit aktivieren",
+        "FIELD_HEADER_MAX_CONNECTIONS"          : "Maximale Anzahl Verbindungen:",
+        "FIELD_HEADER_MAX_CONNECTIONS_PER_USER" : "Maximale Anzahl Verbindungen pro Benutzer:",
+        "SECTION_HEADER_CONCURRENCY" : "Verbindungslimit (Verteilungsgruppe)"
+
+    },
+
+    "DATA_SOURCE_MYSQL" : {
+        "NAME" : "MySQL"
```
[the remark this answers is not in the document]
Comment (berndhose) [reply extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/resources/translations/de.json]: translation is ok

File: extensions/guacamole-auth-openid/src/main/resources/translations/de.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
+
+    "DATA_SOURCE_OPENID" : {
+        "NAME" : "OpenID SSO Backend"
```
[the remark this answers is not in the document]
Comment (berndhose) [reply extensions/guacamole-auth-openid/src/main/resources/translations/de.json]: translation is ok

File: extensions/guacamole-auth-quickconnect/src/main/resources/translations/de.json [hunk 8]
```diff
@@ -0,0 +1,18 @@
+{
+
+    "DATA_SOURCE_QUICKCONNECT" : {
+        "NAME" : "QuickConnect"
```
[the remark this answers is not in the document]
Comment (berndhose) [reply extensions/guacamole-auth-quickconnect/src/main/resources/translations/de.json]: translation is ok

File: extensions/guacamole-auth-radius/src/main/resources/translations/de.json [hunk 9]
```diff
@@ -0,0 +1,13 @@
+{
+
+    "DATA_SOURCE_RADIUS" : {
+        "NAME" : "RADIUS Backend"
```
[the remark this answers is not in the document]
Comment (berndhose) [reply extensions/guacamole-auth-radius/src/main/resources/translations/de.json]: translation is ok

File: extensions/guacamole-auth-totp/src/main/resources/translations/de.json [hunk 10]
```diff
@@ -0,0 +1,34 @@
+{
+
+    "DATA_SOURCE_TOTP" : {
+        "NAME" : "TOTP TFA Backend"
```
[the remark this answers is not in the document]
Comment (berndhose) [reply extensions/guacamole-auth-totp/src/main/resources/translations/de.json]: translation is ok

File: guacamole/src/main/webapp/translations/de.json [hunk 11]
```diff
@@ -336,47 +437,80 @@
...
+        "FIELD_OPTION_SERVER_LAYOUT_TR_TR_QWERTY" : "Türkisch (Qwerty)",
 
         "NAME" : "RDP",
 
         "SECTION_HEADER_AUTHENTICATION"     : "Authentifizierung",
         "SECTION_HEADER_BASIC_PARAMETERS"   : "Basiseinstellungen",
         "SECTION_HEADER_DEVICE_REDIRECTION" : "Geräteumleitung",
         "SECTION_HEADER_DISPLAY"            : "Bildschirm",
+        "SECTION_HEADER_GATEWAY"            : "Remote Desktop Gateway",
+        "SECTION_HEADER_LOAD_BALANCING"     : "Load Balancing",
```
[the remark this answers is not in the document]
Comment (berndhose) [reply guacamole/src/main/webapp/translations/de.json]: translation is ok

File: guacamole/src/main/webapp/translations/de.json (hunk 1 above)
Comment (berndhose) [reply guacamole/src/main/webapp/translations/de.json]: translation is ok

File: guacamole/src/main/webapp/translations/de.json (hunk 2 above)
Comment (berndhose) [reply guacamole/src/main/webapp/translations/de.json]: Should be "Terminalverhalten"

File: guacamole/src/main/webapp/translations/de.json [hunk 12]
```diff
@@ -4,29 +4,36 @@
     
     "APP" : {
 
+        "NAME"    : "Apache Guacamole",
+        "VERSION" : "${project.version}",
```
[the remark this answers is not in the document]
Comment (leetxyz) [reply guacamole/src/main/webapp/translations/de.json]: Thanks, omitted.

File: guacamole/src/main/webapp/translations/de.json [hunk 13]
```diff
@@ -261,23 +292,74 @@
         "ERROR_PASSWORD_MISMATCH" : "@:APP.ERROR_PASSWORD_MISMATCH",
 
         "FIELD_HEADER_ADMINISTER_SYSTEM"             : "Administration:",
-        "FIELD_HEADER_CHANGE_OWN_PASSWORD"           : "Ändere eigenes Passwort:",
-        "FIELD_HEADER_CREATE_NEW_USERS"              : "Erstelle neue Benutzer:",
-        "FIELD_HEADER_CREATE_NEW_CONNECTIONS"        : "Erstelle neue Verbindung:",
-        "FIELD_HEADER_CREATE_NEW_CONNECTION_GROUPS"  : "Erstelle neue Verbindungsgruppe:",
+        "FIELD_HEADER_CHANGE_OWN_PASSWORD"           : "Eigenes Passwort ändern:",
+        "FIELD_HEADER_CREATE_NEW_USERS"              : "Neuen Benutzer erstellen:",
+		"FIELD_HEADER_CREATE_NEW_USER_GROUPS"        : "Neue Benutzergruppe erstellen:",
```
[the remark this answers is not in the document]
Comment (leetxyz) [reply guacamole/src/main/webapp/translations/de.json]: Thanks, should be fixed.

File: guacamole/src/main/webapp/translations/de.json (hunk 2 above)
Comment (leetxyz) [reply guacamole/src/main/webapp/translations/de.json]: Thanks, fixed.

File: extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/resources/translations/de.json (hunk 3 above)
Comment (leetxyz) [reply extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/resources/translations/de.json]: Thanks, fixed.

**Comment by leetxyz** [comment]
Dear @berndhose 

Thank you for your support, went through all comments and should have fixed them all and updated the PR.

**Comment by mike-jumper** [comment]
The only issues I see are:

1. The entries within the `translations` array in `guac-manifest.json` should be maintained in alphabetical order.
2. Indentation should use spaces, not tabs (the new lines in the various `guac-manifest.json` have been indented with tabs)

Other than the above, looks good to me.

**Comment by leetxyz** [comment]
>     1. The entries within the `translations` array in `guac-manifest.json` should be maintained in alphabetical order.

Should be fixed.

>     2. Indentation should use spaces, not tabs (the new lines in the various `guac-manifest.json` have been indented with tabs)

Should be fixed.

I also found missing de.json references in auth-jdbc and auth-ldap and added them.

**Comment by mike-jumper** [comment]
Looks great to me.

**Pull request merged**